ImageGallery/src/org/sleuthkit/autopsy/imagegallery/ImageGalleryController.java

```diff
@@ -783,21 +865,28 @@ protected void cleanup(boolean success) {
         }
 
         @Override
-        void processFile(AbstractFile f, DrawableDB.DrawableTransaction tr) {
+        void processFile(AbstractFile f, DrawableDB.DrawableTransaction tr) throws TskCoreException {
             final boolean known = f.getKnown() == TskData.FileKnown.KNOWN;
 
             if (known) {
                 taskDB.removeFile(f.getId(), tr);  //remove known files
             } else {
 
                 try {
-                    if (FileTypeUtils.hasDrawableMIMEType(f)) {  //supported mimetype => analyzed
+                    //supported mimetype => analyzed
+                    if ( null != f.getMIMEType() && FileTypeUtils.hasDrawableMIMEType(f)) {
                         taskDB.updateFile(DrawableFile.create(f, true, false), tr);
-                    } else { //unsupported mimtype => analyzed but shouldn't include
+                    }
+                    else { //unsupported mimtype => analyzed but shouldn't include
+                        
+                        // if mimetype of the file hasn't been ascertained, ingest might not have completed yet.
+                        if (null == f.getMIMEType()) {
+                            this.setTaskCompletionStatus(false);
+                        }
```

Comment:
The logic has changed here, I suspect incorrectly. If the MIME type is not set, we should not remove the file from the drawable database, since we will want to process it later when ingest completes.